core-bundle/src/EventListener/SwitchUserListener.php

```diff
@@ -59,9 +60,25 @@ public function __invoke(SwitchUserEvent $event): void
             $targetUser = $targetUser->getUsername();
         }
 
-        $this->logger->info(
-            sprintf('User "%s" has switched to user "%s"', $sourceUser, $targetUser),
-            ['contao' => new ContaoContext(__METHOD__, ContaoContext::ACCESS, $sourceUser)]
-        );
+        $originalUser = null;
+
+        if ($token instanceof SwitchUserToken) {
+            $originalUser = $token->getOriginalToken()->getUser()->getUsername();
+        }
+
+        if ($originalUser === $targetUser) {
+            $this->logger->info(
+                sprintf('User "%s" has quit the impersonation of user "%s"', $originalUser, $sourceUser),
+                ['contao' => new ContaoContext(__METHOD__, ContaoContext::ACCESS, $originalUser)]
+            );
+        } else {
```

Comment:
I would combine the `else` and following `if` here: 
```suggestion
        } elseif (!empty($originalUser)) {
```

Reply:
I think that is not good, because the `if` is only used to reduce the number of loggings. @bytehead How should this look?

Reply:
Sorry, nevermind. I got distracted with the code view and the conversations 🙈